pkg/test/compute/cancel_job_test.go

```diff
@@ -0,0 +1,119 @@
+//go:build integration || !unit
+
+package compute
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+	"github.com/bacalhau-project/bacalhau/pkg/devstack"
+	"github.com/bacalhau-project/bacalhau/pkg/docker"
+	dockermodels "github.com/bacalhau-project/bacalhau/pkg/executor/docker/models"
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/node"
+	"github.com/bacalhau-project/bacalhau/pkg/publicapi/apimodels"
+	clientv2 "github.com/bacalhau-project/bacalhau/pkg/publicapi/client/v2"
+	"github.com/bacalhau-project/bacalhau/pkg/setup"
+	"github.com/bacalhau-project/bacalhau/pkg/test/mock"
+	"github.com/bacalhau-project/bacalhau/pkg/test/scenario"
+	"github.com/bacalhau-project/bacalhau/pkg/test/teststack"
+	nodeutils "github.com/bacalhau-project/bacalhau/pkg/test/utils/node"
+	"github.com/stretchr/testify/suite"
+)
+
+type CancelJobTestSuite struct {
+	suite.Suite
+}
+
+func TestCancelJobTestSuite(t *testing.T) {
+	suite.Run(t, new(CancelJobTestSuite))
+}
+
+func (s *CancelJobTestSuite) TestJobCancellation() {
+	ctx := context.Background()
+
+	docker.MustHaveDocker(s.T())
+
+	fsr, c := setup.SetupBacalhauRepoForTesting(s.T())
+
+	computeConfig, err := node.NewComputeConfigWith(c.Node.ComputeStoragePath, node.ComputeConfigParams{
+		TotalResourceLimits: models.Resources{
+			CPU:    1,
+			Memory: 1 * 1024 * 1024 * 1024,
+			Disk:   1 * 1024 * 1024 * 1024,
+		},
+		IgnorePhysicalResourceLimits: true,
+		ControlPlaneSettings: types.ComputeControlPlaneConfig{
+			ResourceUpdateFrequency: types.Duration(50 * time.Millisecond),
+		},
+	})
+
+	s.Require().NoError(err)
+
+	requesterConfig, err := node.NewRequesterConfigWith(node.RequesterConfigParams{
+		NodeOverSubscriptionFactor: 2,
+	})
+	s.Require().NoError(err)
+
+	stack := teststack.Setup(ctx, s.T(), fsr, c,
+		devstack.WithNumberOfRequesterOnlyNodes(1),
+		devstack.WithNumberOfComputeOnlyNodes(1),
+		devstack.WithComputeConfig(computeConfig),
+		devstack.WithRequesterConfig(requesterConfig),
+	)
+
+	nodeutils.WaitForNodeDiscovery(s.T(), stack.Nodes[0].RequesterNode, 2)
+
+	es, err := dockermodels.NewDockerEngineBuilder("ubuntu").
+		WithEntrypoint("bash", "-c", "sleep 100000").
+		Build()
```

Comment:
I would be careful about sleeping too much where failures would block or delay the test termination